modules/library/render/src/main/java/org/geotools/renderer/composite/IntegerBgrAccessor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.renderer.composite;
+
+import java.awt.image.Raster;
+import java.awt.image.WritableRaster;
+
+/**
+ * A color component accessor for integer oriented rasters in BGR order
+ * 
+ * @author Andrea Aime - GeoSolutions
+ */
+class IntegerBgrAccessor implements RgbaAccessor {
```

Comment:
JAI does not provide what I need, it works at the sample level, not at the color level.
Implementation wise, JAI has a full copy of the whole operator for performance reasons, not just an accessor class. You can say I've been a bit lazy here.